File: armstrong/armstrong.rb
n=0
sum=0
r=0
puts "enter any number"
 n=gets.chomp.to_i
n1=n
 while n>0
    r=n%10
    sum=sum+(r*r*r)
    n=n/10
 end 
if n1==sum
 puts "entered number is an armstrong number "
else
 puts "entered number is not an armstrong number"
end

File: armstrong/armstrong.c
#include <stdio.h> 
int main()
{
   int n,sum = 0,temp,r;
 
   printf("Enter an integer\n");
   scanf("%d",&n);
 
   temp = n;
 
   while( temp != 0 )
   {
      r = temp%10;
      sum = sum + r*r*r;
      temp = temp/10;
   }
 
   if ( n == sum )
      printf("Entered number is an armstrong number.\n");
   else
      printf("Entered number is not an armstrong number.\n");
 
   return 0;
}
